datafusion/sqllogictest/bin/sqllogictests.rs

```diff
@@ -121,7 +117,7 @@ async fn run_test_file(test_file: TestFile) -> Result<()> {
         relative_path,
     } = test_file;
     info!("Running with DataFusion runner: {}", path.display());
-    let Some(test_ctx) = context_for_test_file(&relative_path).await else {
+    let Some(test_ctx) = TestContext::try_new_for_test_file(&relative_path).await else {
```

Comment:
I moved the logic for creating TestContext into the `datafusion_sqllogictest` library (and out of the runner) to avoid making the runner even longer